src/citrine/resources/branch.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional, Union, Iterator
+from uuid import UUID
+
+from citrine._rest.collection import Collection
+from citrine._rest.resource import Resource, ResourceTypeEnum
+from citrine._serialization import properties
+from citrine._session import Session
+from citrine._utils.functions import format_escaped_url
+from citrine.resources.design_workflow import DesignWorkflowCollection
+from citrine.resources.response import Response
+
+
+class Branch(Resource['Branch']):
+    _response_key = None
+    _resource_type = ResourceTypeEnum.BRANCH
+
+    name = properties.String('name')
+    uid = properties.Optional(properties.UUID(), 'id')
+    created_at = properties.Optional(properties.Datetime(), 'created_at', serializable=False)
+    updated_at = properties.Optional(properties.Datetime(), 'updated_at', serializable=False)
+
+    project_id: Optional[UUID] = None
+
+    def __init__(self,
+                 name: str,
+                 *,
+                 session: Optional[Session] = None):
+        self.name: str = name
+        self.session: Session = session
+
+    def __str__(self):
+        return f'<Branch {self.name!r}>'
+
+    def _path(self):
+        return format_escaped_url('/branch/{branch_id}', branch_id=self.uid)
+
+    @property
+    def design_workflows(self) -> DesignWorkflowCollection:
+        """Return a resource representing all workflows contained within this branch."""
+        if getattr(self, 'project_id', None) is None:
+            raise AttributeError('Cannot initialize workflow without project reference!')
+        return DesignWorkflowCollection(project_id=self.project_id, branch_id=self.uid, session=self.session)
+
+
+class BranchCollection(Collection[Branch]):
+    """A collection of Branches."""
+
+    _path_template = '/projects/{project_id}/design-workflow-branches'
+    _individual_key = None
+    _collection_key = 'response'
+    _resource = Branch
+
+    def __init__(self, project_id: UUID, session: Session):
+        self.project_id: UUID = project_id
+        self.session: Session = session
+
+    def build(self, data: dict):
+        branch = Branch.build(data)
+        branch.session = self.session
+        branch.project_id = self.project_id
+        return branch
+
+    def register(self, name: str) -> Branch:
+        """
+        Create and upload new branch.
+
+        Parameters
+        ----------
+        name: str
+            Name of the project to be created.
+
+        """
+        return super().register(Branch(name))
+
+    def list(self, *,
```

Comment:
I think somewhere in the paginator, we've deprecated the option to pass in page as an argument. If that's right, let's not add it here. (if we find we need it, we can add it back in, but no need to pass it in now)

Reply:
Gotcha. I did see somewhere we state it's deprecated, but I figured since it's still there we would need to keep using it until it's gone. I'll remove it.